### PR TITLE
Add atomic units of energy, mass, length, electric charge and current.

### PR DESCRIPTION
**Pull request by crystal-growth**
Adding 

- Dalton (unified atomic mass unit = 1/12 * (mass of C12))
- Hartree (atomic unit of energy, ~27.2 eV)
- atomic unit of length (Bohr radius, 5.291 772 109 03 x 10-11 m )
- atomic unit of charge (absolute charge of electron, 1.602_176_634 x 10^-19 C)

### CODE REVIEW

File: src/si/electric_charge.rs [hunk 1]
```diff
@@ -48,6 +48,8 @@ quantity! {
         @microampere_hour: 3.6_E-3; "µA · h", "microampere hour", "microampere hours";
 
         @abcoulomb: 1.0_E1; "abC", "abcoulomb", "abcoulombs";
+        @atomic_unit_of_charge: 1.602_176_634_E-19; "a.u. of charge",
+            "atomic unit of charge", "atomic units of charge";
```
Comment (iliekturtles) [inline src/si/electric_charge.rs:52]:
```suggestion
        /// Electric charge carried by a single proton.
        @elementary_charge: 1.602_176_634_E-19; "e", "elementary charge", "elementary charges";
        @atomic_unit_of_charge: 1.602_176_634_E-19; "a.u. of charge", "atomic unit of charge",
            "atomic units of charge";
        @abcoulomb: 1.0_E1; "abC", "abcoulomb", "abcoulombs";
```

**Review by iliekturtles — COMMENTED** [review]
I didn't get very far in this review. Comments about using "elementary charge" which seems more common than "atomic units"? Especially since the 2019 redefinition which uses "elementary charge" to define the ampere.

I included a possible change that has both elementary charge and atomic unit of charge.

**Comment by crystal-growth** [comment]
Thanks, completely agree with elementary charge!
Squashed and rebased.

File: src/si/electric_charge.rs [hunk 2]
```diff
@@ -98,6 +102,8 @@ mod tests {
             test::<i::abampere, t::second, q::abcoulomb>();
             test::<i::statampere, t::second, q::statcoulomb>();
 
+            test::<i::atomic_unit_of_charge_per_second, t::second, q::atomic_unit_of_charge>();
+
```
Comment (iliekturtles) [inline src/si/electric_charge.rs:106]: Github won't let me use the full context for the changes, so I had to paste a diff below.

Put the elementary charge and atomic unit of charge tests in the same order as the unit definitions. For elementary charge possibly use elementary charge per second (see electric current review) or fallback to atomic unit of charge per second.

```diff
diff --git a/src/si/electric_charge.rs b/src/si/electric_charge.rs
index c749ec6..ea7406b 100644
--- a/src/si/electric_charge.rs
+++ b/src/si/electric_charge.rs
@@ -98,12 +98,12 @@ mod tests {
             test::<i::zeptoampere, t::second, q::zeptocoulomb>();
             test::<i::yoctoampere, t::second, q::yoctocoulomb>();
 
+            test::<i::elementary_charge_per_second, t::second, q::elementary_charge>();
+            test::<i::atomic_unit_of_charge_per_second, t::second, q::atomic_unit_of_charge>();
             test::<i::ampere, t::hour, q::ampere_hour>();
             test::<i::abampere, t::second, q::abcoulomb>();
             test::<i::statampere, t::second, q::statcoulomb>();
 
-            test::<i::atomic_unit_of_charge_per_second, t::second, q::atomic_unit_of_charge>();
-
             fn test<I: i::Conversion<V>, T: t::Conversion<V>, Q: q::Conversion<V>>() {
                 Test::assert_approx_eq(&ElectricCharge::new::<Q>(V::one()),
                     &(ElectricCurrent::new::<I>(V::one()) * Time::new::<T>(V::one())));
```

File: src/si/electric_current.rs [hunk 3]
```diff
@@ -40,6 +40,8 @@ quantity! {
         @yoctoampere: prefix!(yocto); "yA", "yoctoampere", "yoctoamperes";
 
         @abampere: 1.0_E1; "abA", "abampere", "abamperes";
+        @atomic_unit_of_charge_per_second: 1.602_176_634_E-19; "a.u. of charge/s",
+            "atomic unit of charge per second", "atomic units of charge per second";
```
Comment (iliekturtles) [inline src/si/electric_current.rs:44]: Should elementary charge per second exist? If so, should probably add a comment.

```suggestion
        /// ...
        @elementary_charge_per_second: 1.602_176_634_E-19; "e/s", "elementary charge per second",
            "elementary charges per second";
        @atomic_unit_of_charge_per_second: 1.602_176_634_E-19; "a.u. of charge/s",
            "atomic unit of charge per second", "atomic units of charge per second";
        @abampere: 1.0_E1; "abA", "abampere", "abamperes";
```

File: src/si/energy.rs [hunk 4]
```diff
@@ -56,6 +56,9 @@ quantity! {
         @decaelectronvolt: 1.602_176_634_E-18; "daeV", "decaelectronvolt", "decaelectronvolts";
         @electronvolt: 1.602_176_634_E-19; "eV", "electronvolt", "electronvolts";
 
+        // Atomic unit of energy (Hartree energy)
+        @hartree: 4.359_744_722_2071_E-18; "Ha", "hartree", "hartrees";
```
Comment (iliekturtles) [inline src/si/energy.rs:60]: `Eₕ` seems more common than `Ha`. Thoughts?

```suggestion
        /// Atomic unit of energy (Hartree energy).
        @hartree: 4.359_744_722_207_1_E-18; "Eₕ", "hartree", "hartrees";
```

File: src/si/length.rs [hunk 5]
```diff
@@ -39,6 +39,8 @@ quantity! {
         @yoctometer: prefix!(yocto); "ym", "yoctometer", "yoctometers";
 
         @angstrom: 1.0_E-10; "Å", "ångström", "ångströms";
+        // Atomic unit of length (Bohr radius)
+        @atomic_unit_of_length: 5.291_772_109_03_E-11; "a.u. of length", "atomic unit of length", "atomic units of length";
```
Comment (iliekturtles) [inline src/si/length.rs:43]: Is Bohr radius actually used as a unit?

```suggestion
        @bohr_radius: 5.291_772_109_03_E-11; "a₀", "bohr radius", "bohr radiuses";
        /// Atomic unit of length (Bohr radius).
        @atomic_unit_of_length: 5.291_772_109_03_E-11; "a.u. of length", "atomic unit of length",
            "atomic units of length";
```

File: src/si/mass.rs [hunk 6]
```diff
@@ -39,7 +39,8 @@ quantity! {
         @zeptogram: prefix!(zepto) / prefix!(kilo); "zg", "zeptogram", "zeptograms";
         @yoctogram: prefix!(yocto) / prefix!(kilo); "yg", "yoctogram", "yoctograms";
 
-        @carat: 2.0_E-4; "ct", "carat", "carats";
+        @carat: 2.0_E-4; "ct", "carat", "carats";   
+        @dalton: 1.660_539_066_60_E-27; "Da", "dalton", "daltons"; // Atomic mass unit
```
Comment (iliekturtles) [inline src/si/mass.rs:43]:
```suggestion
        @carat: 2.0_E-4; "ct", "carat", "carats";
        /// Unified atomic mass unit.
        @dalton: 1.660_539_066_60_E-27; "Da", "dalton", "daltons";
```

**Review by iliekturtles — CHANGES_REQUESTED** [review]
Couple more questions about some of the constants that seem to be used as units. Otherwise close to merging!

**Comment by crystal-growth** [comment]
Thanks a lot for these suggestions!

1) Added elementary charge per second. It definitely should exist.
2) Agree with Eₕ, it's more common, and better stresses that it is Energy
3) I'm not sure if Bohr radius is an "official" unit, but it is very convenient to use it for describing hydrogen-like states in semiconductor physics:
`
let bohr_radius_of_shallow_donor_in_silicon = Length::new::<bohr_radius>(some_nondimentional_scaling_factor);
`
so I fully agree with adding it together with a.u. of length.

File: src/si/electric_current.rs [hunk 7]
```diff
@@ -39,6 +39,11 @@ quantity! {
         @zeptoampere: prefix!(zepto); "zA", "zeptoampere", "zeptoamperes";
         @yoctoampere: prefix!(yocto); "yA", "yoctoampere", "yoctoamperes";
 
+        /// Atomic unit of charge (electric charge carried by a single proton) per second
```
Comment (iliekturtles) [inline src/si/electric_current.rs:42]:
```suggestion
        /// Atomic unit of charge (electric charge carried by a single proton) per second.
```

**Comment by iliekturtles** [comment]
Thanks so much for this PR! I just returned from vacation and will get back to working on the rest of your PRs.

**Comment by crystal-growth** [comment]
Thank you very much! I apologize for this heap of PRs, and hope it will help improving usability of this great library. 

Starting using uom, I found it extremely useful in protecting me from various errors, including using wrong dimension for a given quantity. 
However, if a needed quantity is missing -- it becomes much less ergonomic. 

Now I'm "dimensionalizing" all my computational scripts, adding PRs for the quantities and units that are missing.
It's a great pleasure to get rid of functions with signatures like:
`some_func(atom_radius_in_picometers: f64 , atom_mass_in_amu: f64) -> diffusion_coeffient_in_cm2_s:f64`

**Pull request merged**